packages/boxel-ui/addon/src/components/tabbed-header/usage.gts

```diff
@@ -0,0 +1,141 @@
+import { fn } from '@ember/helper';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import FreestyleUsage from 'ember-freestyle/components/freestyle/usage';
+
+import { ALL_ICON_COMPONENTS } from '../../icons.gts';
+import Icon from '../../icons/sparkle.gts';
+import TabbedHeader from './index.gts';
+
+export default class TabbedHeaderUsage extends Component {
+  @tracked tabs = [
+    {
+      displayName: 'Dashboard',
+      tabId: 'dashboard',
+    },
+    {
+      displayName: 'Requirements',
+      tabId: 'requirements',
+    },
+    {
+      displayName: 'Your Apps',
+      tabId: 'your-apps',
+    },
+    {
+      displayName: 'Sample Apps',
+      tabId: 'sample-apps',
+    },
+    {
+      displayName: 'Favorites',
+      tabId: 'favorites',
+    },
+  ];
+  @tracked title = 'AI App Generator';
+  @tracked iconComponent = Icon;
+  @tracked iconURL = '';
+  @tracked activeTabIndex = 0;
+  @tracked headerColor = '#ffd800';
+  @tracked iconBackgroundColor?: string;
+  @tracked iconBorderColor?: string;
+  @tracked iconCoversAllAvailableSpace = false;
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    let index = this.tabs?.findIndex(
+      (tab) => tab.tabId === window.location?.hash?.slice(1),
+    );
+    if (index && index !== -1) {
+      this.activeTabIndex = index;
+    }
+  }
+
+  onSetActiveTab = (index: number) => {
+    this.activeTabIndex = index;
+  };
+
+  <template>
+    <FreestyleUsage @name='TabbedHeader'>
+      <:example>
+        <TabbedHeader
+          @title={{this.title}}
+          @tabs={{this.tabs}}
+          @onSetActiveTab={{fn this.onSetActiveTab}}
+          @activeTabIndex={{this.activeTabIndex}}
+          @iconComponent={{this.iconComponent}}
+          @iconURL={{this.iconURL}}
+          @headerBackgroundColor={{this.headerColor}}
+          @iconBackgroundColor={{this.iconBackgroundColor}}
+          @iconBorderColor={{this.iconBorderColor}}
+          @iconCoversAllAvailableSpace={{this.iconCoversAllAvailableSpace}}
+        />
```

Comment:
With all these icon-related properties, might it be cleaner to use a named `:icon` block or the like to provide customisation?

Reply:
making a named block cleaned it up nicely. added a transition, not the same animation, but smoother than before.

Reply:
nice 🎉